src/manifest/factory.ts

```diff
@@ -58,7 +61,21 @@ export default function createManifest(
   warning$ : Subject<Error|CustomError>
 ) : Manifest {
   manifestObject.periods = (manifestObject.periods).map((period) => {
-    period.adaptations = checkAdaptations(period.adaptations, warning$);
+    (Object.keys(period.adaptations) as IAdaptationType[]).forEach((type) => {
+      const adaptationsForType = period.adaptations[type];
+      if (adaptationsForType) {
```

Comment:
This condition should be always true (from TS typings)

Reply:
It seems not. All typed adaptations can be undefined.